canals/pipeline/validation.py

```diff
@@ -6,13 +6,40 @@
 
 import networkx
 
+from canals.utils import _type_name
 from canals.errors import PipelineValidationError
 from canals.component.sockets import InputSocket, OutputSocket
 
 
 logger = logging.getLogger(__name__)
 
 
+def _describe_pipeline_inputs(graph: networkx.MultiDiGraph):
+    """
+    Returns a dictionary with the input names and types that this pipeline accepts.
+    """
+    inputs = {
+        comp: {socket.name: {"type": socket.type, "is_optional": socket.is_optional} for socket in data}
+        for comp, data in _find_pipeline_inputs(graph).items()
+        if data
+    }
+    return inputs
```

Comment:
This is not validation, I don't see why it should be part of this module.

Reply:
I created another module called `descriptions` (for lack of a better idea) when I moved the `describe` and `find` methods. The `find` methods kinda must be there, to avoid a circular dependency issue